src/pages/Account/Confirm.tsx

```diff
@@ -16,104 +16,92 @@ import * as DashboardText from '../../features/Dashboard/DashboardText';
-      >
-        <Helmet>
-          <title>Confirmation | {HACKATHON_NAME}</title>
-        </Helmet>
-        <MaxWidthBox
-          hidden={this.state.wasConfirmed && !this.state.attempting}
-          mb={'20px'}
-        >
-          <Image src={constructionCone} imgHeight={'6rem'} />
-        </MaxWidthBox>
-        <Box style={{ marginTop: '6rem' }}>
-          <H1 fontSize={'40px'}>{result}</H1>
-        </Box>
-        <MaxWidthBox hidden={this.state.attempting} mb={'20px'}>
-          <Paragraph
-            fontSize={'24px'}
-            maxWidth={'600px'}
-            marginLeft={'16px'}
-            marginRight={'16px'}
-            textAlign={'center'}
-          >
-            {paragraphMessage}
-          </Paragraph>
-        </MaxWidthBox>
-        <MaxWidthBox hidden={this.state.attempting}>
-          <Link to={link}>
-            <Button>{buttonMessage}</Button>
-          </Link>
-        </MaxWidthBox>
-      </Flex>
-    );
-  }
+  // Is the account confirmed?
+  const [wasConfirmed, setWasConfirmed] = useState<boolean>(false);
 
-  public async componentDidMount() {
-    try {
-      const token = getTokenFromQuery();
-      const response = await Auth.confirm(token);
-      if (response.status === 200) {
-        console.log('Confirmed account');
-        this.setState({
-          attempting: false,
-          wasConfirmed: true,
-        });
-      } else {
-        this.setState({
-          attempting: false,
-        });
-        console.error('Did not confirm account');
+  // When component mounts, fetch account info on if the account is confirmed
+  // from the api, then when finished set attempting to false
+  useEffect(() => {
+    (async () => {
+      try {
+        const token = getTokenFromQuery();
+        const response = await Auth.confirm(token);
+        alert(response);
+        if (response.status === 200) {
+          console.log('Confirmed account');
+          setAttempting(false);
+          setWasConfirmed(true);
+        } else {
+          setAttempting(false);
+          setWasConfirmed(false);
+          console.error('Did not confirm account');
+        }
+      } catch (e) {
+        setAttempting(true);
+        setWasConfirmed(false);
+        console.error('No token found in the query parameters');
       }
-    } catch (e) {
-      this.setState({
-        attempting: false,
-      });
-      console.error('No token found in the query parameters');
-    }
+    })();
+  }, []);
+
+  let result;
+  let paragraphMessage;
+  let buttonMessage;
+  let link;
+  if (wasConfirmed) {
+    result = DashboardText.ConfirmAccount;
+    paragraphMessage = DashboardText.ConfirmMessage;
+    buttonMessage = DashboardText.Continue;
+    link = DashboardText.ConfirmLink;
+  } else if (attempting) {
+    result = DashboardText.UnableConfirm;
+    paragraphMessage = DashboardText.Error;
+    buttonMessage = DashboardText.CreateAccount;
+    link = DashboardText.AttemptingLink;
+  } else {
```

Comment:
Instead of this final else block, why not just set the default values of result, paragraphMessage, etc to be the values listed in this last else block?

Reply:
Good point, that's cleaner code